src/workloads/scale/UpdateSingleLargeDocumentWith10kThreads.yml

```diff
@@ -0,0 +1,69 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: >
+  The workload inserts a single large document, and tests concurrently updating the same document
+  with many threads. This is intended to test the behavior of the server under heave memory pressure
+  caused by update operations with a high rate of write conflicts. The update operation is on an
+  integer field, the command itself is relatively small, so most of the memory pressure should come
+  from the query subsystem.
+
+GlobalDefaults:
+  MaxPhases: &MaxPhases 2
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 10100
+
+Actors:
+
+# Phases:
+# 1. Insert a single document close to the 16MiB limit.
+# 2. Use 10k threads attempting to update the same document.
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Database: &DB test
+        Repeat: 1
+        Threads: 1
+        CollectionCount: 1  # Collection name will be Collection0, this is not configurable.
+        DocumentCount: 1
+        BatchSize: 1
+        Document:
+          _id: 0
+          data: {^FastRandomString: {length: 16000000}}
+          a: 0
+
+- Name: PerformUpdates
+  Type: CrudActor
+  Database: *DB
+  Threads: 10000
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 3 minutes
```

Comment:
My understanding is that either `Duration` or `Repeat` must be specified. There's another config which is `Blocking: None`, but that one depends on at least one of the actors of the Phase specifying a Duration or Repeat.

Apparently using it currently kind of works, but the documentation warns against it.

> If neither `Duration` nor `Repeat` is given for a `Phase` block,
the YAML must specify `Blocking: None` to prevent accidentally
causing an Actor to not run at all during a Phase. It is still possible
for all Actors to specify `Blocking:None` for a given Phase in which
case the number of iterations that all Actors will take in that Phase
is undefined (but in practice is either 0 or 1 depending on how fast
all the Actors call into the `Orchestrator` to say they're done blocking
in the current Phase.)

>   2. It is undefined behavior if no Actors specify a Duration
     or Repeat for a Phase. It does not currently cause an
     infinite-loop, but we don't check for this situation
     (which is almost certainly a mis-configuration) at setup
     time.

Reply:
I will set it to 10 minutes, but currently in it takes less than 10 seconds to OOM with this config:
```
mongodb_setup:
  name: single-replica
  release: 2022-11
```